modules/sdk-lib-mpc/src/tss/ecdsa-dkls/dkg.ts

```diff
@@ -1,22 +1,70 @@
 import { KeygenSession, Keyshare, Message } from '@silencelaboratories/dkls-wasm-ll-node';
-import { DeserializedBroadcastMessage, DeserializedMessages, DkgState } from './types';
+import { DeserializedBroadcastMessage, DeserializedMessages, DkgState, RetrofitData } from './types';
 import { decode } from 'cbor';
+import { encode } from 'cbor-x';
+import { bigIntToBufferBE } from '../../util';
+import { Secp256k1Curve } from '../../curves';
 
 export class Dkg {
   protected dkgSession: KeygenSession | undefined;
   protected dkgSessionBytes: Uint8Array;
   protected dkgKeyShare: Keyshare;
+  protected keyShareBuff: Buffer;
   protected n: number;
   protected t: number;
   protected chainCodeCommitment: Uint8Array | undefined;
   protected partyIdx: number;
   protected dkgState: DkgState = DkgState.Uninitialized;
+  protected dklsKeyShareRetrofitObject: Keyshare | undefined;
 
-  constructor(n: number, t: number, partyIdx: number) {
+  constructor(n: number, t: number, partyIdx: number, retrofitData?: RetrofitData) {
     this.n = n;
     this.t = t;
     this.partyIdx = partyIdx;
     this.chainCodeCommitment = undefined;
+    if (retrofitData) {
+      if (!retrofitData.xShare.y || !retrofitData.xShare.chaincode || !retrofitData.xShare.x) {
+        throw Error('xShare must have a public key, private share value, and a chaincode.');
+      }
+      if (n !== 3 || t !== 2) {
+        throw Error('only 2 of 3 retrofit is supported.');
+      }
+      if (retrofitData.bigSiList.length !== 2) {
+        throw Error("bigSiList should contain the other parties' Si's");
+      }
```

Comment:
these conditions can be extracted to a function

Reply:
not sure the validation needs to be in it's own function since it's very minimal and also it is needed there to assert that xShare.y, xShare.chaincode, xShare.x are defined so that the compiler does not complain when we later use them as defined only variables later in the same code block.